common/djangoapps/student/views.py

```diff
@@ -2314,4 +2325,8 @@ def _get_course_programs(user, user_enrolled_courses):  # pylint: disable=invali
             except KeyError:
                 log.warning('Program structure is invalid, skipping display: %r', program)
 
+    # If programs_data is empty dict even than set the cache
+    if is_cache_enabled_on_dashboard_programs():
```

Comment:
This condition is not doing what the above comment is describing. Please correct the comment or update the condition.